pkgs/servers/x11/xorg/overrides.nix

```diff
@@ -1,109 +1,725 @@
+      (lib.mesonEnable "zlib" true)
+    ];
+
+    meta = attrs.meta // {
+      license = with lib.licenses; [
+        mit
+        isc
+        x11
+      ];
+      # https://gitlab.freedesktop.org/xorg/lib/libpciaccess/-/blob/master/configure.ac#L108-114
+      platforms = lib.fold (os: ps: ps ++ lib.platforms.${os}) [ ] [
+        "cygwin"
+        "freebsd"
+        "linux"
+        "netbsd"
+        "openbsd"
+        "illumos"
+      ];
+      badPlatforms = [
+        # mandatory shared library
+        lib.systems.inspect.platformPatterns.isStatic
+      ];
+    };
+  });
+
+  libpthreadstubs = prev.libpthreadstubs.overrideAttrs (attrs: {
+    meta = attrs.meta // {
+      license = lib.licenses.x11DistributeModifications; # but with missing copyright notice
+      # only contains a pkgconfig file on linux and windows
+      platforms = lib.platforms.unix ++ lib.platforms.windows;
+    };
+  });
+
+  libSM = prev.libSM.overrideAttrs (attrs: {
+    outputs = [
+      "out"
+      "dev"
+      "doc"
+    ];
+    propagatedBuildInputs = attrs.propagatedBuildInputs or [ ] ++ [ xorg.libICE ];
+    meta = attrs.meta // {
+      license = with lib.licenses; [
+        mit
+        mitOpenGroup
+      ];
+    };
+  });
+
+  libWindowsWM = prev.libWindowsWM.overrideAttrs (attrs: {
+    configureFlags = attrs.configureFlags or [ ] ++ malloc0ReturnsNullCrossFlag;
+    meta = attrs.meta // {
+      license = lib.licenses.mit;
+    };
+  });
+
+  libX11 = prev.libX11.overrideAttrs (attrs: {
+    outputs = [
+      "out"
+      "dev"
+      "man"
+    ];
+    configureFlags = attrs.configureFlags or [ ] ++ malloc0ReturnsNullCrossFlag;
+    depsBuildBuild =
+      [ buildPackages.stdenv.cc ]
+      ++ lib.optionals stdenv.hostPlatform.isStatic [
+        (xorg.buildPackages.stdenv.cc.libc.static or null)
+      ];
+    preConfigure = ''
+      sed 's,^as_dummy.*,as_dummy="\$PATH",' -i configure
+    '';
+    postInstall = ''
+      # Remove useless DocBook XML files.
+      rm -rf $out/share/doc
+    '';
+    CPP = lib.optionalString stdenv.isDarwin "clang -E -";
+    propagatedBuildInputs = attrs.propagatedBuildInputs or [ ] ++ [ xorg.xorgproto ];
+    meta = attrs.meta // {
+      license = with lib.licenses; [
+        mitOpenGroup
+        x11
+        mit
+        hpndSellVariant
+        hpnd
+        smlnj
+        bsd1
+        isc
+        hpndDocSell
+        # some unknown permission notices that grant all 4 freedoms
+        free
+        # quite a few files carry this copyright notice:
+        # > (c) Copyright 1995 FUJITSU LIMITED
+        # > This is source code modified by FUJITSU LIMITED under the Joint
+        # > Development Agreement for the CDE/Motif PST.
+        # commit 5e7d589697755a70fb22d85c6a1ae82b39843e53 removes 2 files
+        # with the same copyright notice and states:
+        # > This is unpublished proprietary source code of FUJITSU LIMITED
+        unfree
```

Comment:
I just did a grep on the source we are pulling, and the only reference to `unpublished proprietary` I can find is in the Changelog. Maybe verify from what source you're checking.

Reply:
Yes, "This is unpublished proprietary source code of FUJITSU LIMITED" is only part of the changelog.
My assumption was that this applies to all files that have the same Copyright statement.

Reply:
The removed Fujitsu files have the following copyright notice:

```
XCOMM Copyright 1995 by FUJITSU LIMITED
XCOMM This is source code modified by FUJITSU LIMITED under the Joint
XCOMM Development Agreement for the CDEnext PST.
XCOMM This is unpublished proprietary source code of FUJITSU LIMITED
```

The ones still present in the libX11 tree have this one:

```
 * Copyright 1995 by FUJITSU LIMITED
 * This is source code modified by FUJITSU LIMITED under the Joint
 * Development Agreement for the CDE/Motif PST.
```

This differs both in omitting the questioned “This is unpublished proprietary source code of FUJITSU LIMITED” notice and in the name of the cited Joint Development Agreement (though presumably there is some relation). I checked a few files with these notices and they all were in files that already contained FOSS licence terms attached to a Toshiba copyright notice (perhaps their partner in the Joint Development Agreement?). Therefore, I believe they function as simple notices of Fujitsu’s copyright in the files, rather than being indications of a lack of accompanying copyright licence. I don’t think there is any reason to believe that this code is non‐free, but I would again suggest raising it with upstream first if you have concerns about it.

Reply:
oh i kind of assumed the last sentence was a commentary by the commiter
dumb of me

Reply:
https://gitlab.freedesktop.org/xorg/lib/libx11/-/issues/217

Reply:
In my view, there’s no reason to treat this differently to e.g.

```c
/*
 * Copyright 2024 Author 1
 * Copyright 2024 Author 2
 *
 * […standard MIT licence text…]
 */
```

It’s true that the licence text says TOSHIBA specifically, but only in terms of e.g. non‐advertisement and warranty disclaimer, not in terms of the basic copyright grant. The copyright notice being in a separate comment block, without any statement of differing licence terms or reservation of rights, doesn’t indicate any negation of the licence terms.

You could object that there’s no explicit statement that Fujitsu also releases its code under that licence, but that also applies to the very conventional example I gave – whether the owner of the copyright releases their work under the licence isn’t really determined by that, although I would consider contributing to a file with an existing FOSS licence notice without explicitly reserving rights to be an indicator of intent in that regards. For instance, GitHub pull requests like this one are assumed to be licensed under the project’s licence despite usually not containing any notice of such because the Terms of Service obligate the submitting user to agree to this (and also presumably because the intent is also just obvious, regardless of your feelings on the enforceability of website ToSes). I presume that there were agreements between Toshiba and Fujitsu that lead to it being released under a FOSS licence to begin with – either at the time it was written or as part of a later opening up of the code.

I think under the general presumption (that we can’t avoid in a huge distribution project such as this) that code which says it is licensed a certain way should be treated that way unless there is strong reason to believe otherwise, there’s no cause for concern here, especially given that we know that Fujitsu did explicitly put a proprietary disclaimer on other files that they may not have wanted released. (Although, even then I sort of suspect that it was just a hold‐over notice that they forgot to remove when deciding to release code, and only removed from the modern codebase out of caution because the pre‐Unicode locale files were obsolete anyway.) Maybe someone on the X.org side knows the full history here about what was going on 30 years ago though…